### PR TITLE
fix: fix PQ training metric type is not appropriately propogated

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

**Review by chebbyChefNEQ — APPROVED** [review]

File: python/python/lance/dataset.py [hunk 1]
```diff
@@ -827,6 +827,7 @@ def create_index(
             "dot",
         ]:
             raise ValueError(f"Metric {metric} not supported.")
+        kwargs["metric"] = metric.lower()
```
Comment (westonpace) [inline python/python/lance/dataset.py:830]: Why is this needed if we pass `metric` to `create_index` down below?

**Review by westonpace — APPROVED** [review]

File: python/python/lance/dataset.py (hunk 1 above)
Comment (eddyxu) [reply python/python/lance/dataset.py:830]: Good catch. This was a left-over from debugging. I will file another PR to fix this one.
